flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,540 @@
+            'include_available': {'type': 'boolean', 'default': False},
+            'include_year': {'type': 'boolean', 'default': False},
+            'include_ep_title': {'type': 'boolean', 'default': False},
+        },
+        'oneOf': [{'required': ['username', 'password']}, {'required': ['api_key']}],
+        'required': ['base_url', 'type'],
+        'additionalProperties': False,
+    }
+
+    @property
+    def immutable(self):
+        return False
+
+    def __init__(self, config: Dict[str, Any]):
+        self.config = config
+        self._items = None
+        self._cached_items = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def add(self, entry: Entry):
+
+        log.info(f"Adding {entry['title']} to ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+
+        log.info(f"{entry['title']} was added to ombi_list.")
+
+    def __ior__(self, entries: List[Entry]):
+
+        for entry in entries:
+            self.add(entry)
+
+    def discard(self, entry: Entry):
+        """Removes an entry from OMBI by marking it as available.
+
+        Args:
+            entry (Entry): An item from a task.
+        """
+
+        log.info(f"Removing {entry['title']} from ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+        ombi_entry.mark_available()
+
+        log.info(f"{entry['title']} was removed from ombi_list.")
+
+    def __isub__(self, entries: List[Entry]):
+
+        for entry in entries:
+            self.discard(entry)
+
+    def _find_entry(self, entry: Entry):
+
+        if "tmdb_id" not in entry:
+            log.warning(
+                f"{entry['title']} is missing the tmdb_id, consider using tmdb_lookup plugin."
+            )
+
+        for item in self.items:
+            if item['tmdb_id'] == entry['tmdb_id']:
+                return item
+
+        return None
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    def clear(self):
+        if self.items:
+            for item in self.items:
+                self.discard(item)
+            self._cached_items = None
+
+    def get(self, entry):
+        return self._find_entry(entry)
+
+    @property
+    def items(self) -> List[Entry]:
+        if not self._items:
+
+            json = self.get_request_list()
+
+            self._items = []
+
+            for parent_request in json:
+                if self.config.get('type') == 'movies':
```

Comment:
Might be cleaner to split movies/series logic into own methods

Reply:
Agreed, but not my code and I'm worried about touching it.

Reply:
> Agreed, but not my code and I'm worried about touching it.

Be brave, it usually pays out :)

And with git, it's easy to back out if not

Reply:
I do this in my personal time like most of the contributors. I'm not going to waste my free time making working code "cleaner". 

I was already brave when I stepped up to finish this PR. You could also do the same?

Reply:
No sweat, just trying to put out some encouragement :)

All contributions are really appreciated. This project would not be alive without them today. 

I think this PR is good enough aside from the "base_url" / "port" issue. We don't want to pile more configuration changes to #3507 which will affect the end users.

Reply:
I will focus on this some soon, get these changes in place and then do some testing with my setup. Then we can get this merged and get some feedback on it from real users.

Reply:
Will have to refactor later, perhaps after we have tests.